pkg/cacheutil/memcached_client.go

```diff
@@ -441,6 +453,45 @@ func (c *memcachedClient) GetMulti(ctx context.Context, keys []string) map[strin
 	return hits
 }
 
+func (c *memcachedClient) Del(ctx context.Context, key string) error {
+	errCh := make(chan error, 1)
+
+	// Enqueue delete operation to ensure the affected item is effectively removed,
+	// even when prior set operations are still waiting in the async queue.
+	err := c.enqueueAsync(func() {
```

Comment:
Queue was thought to be used for async operations. Why using the queue at all for a sync operation?

Reply:
> Why using the queue at all for a sync operation?

This is indeed a synchronous operation. However, we want to wait until any previous set op. affecting the same key has been processed before doing the invalidation.

And here's the why: 

```go
// set op. enqueued for k1
c.SetAsync(ctx, "k1", data, time.Hour)  

// delete op. is not effective, as previous set op. is still enqueued...
c.Delete(ctx, "k1")

// ... set op. for "k1" is processed
```

Just trying to be consistent here, as IMO this may lead to a wide variety of issues that could be really hard to track later in production.